emmet-core/emmet/core/qchem/calc_types/calc_types.py

```diff
@@ -0,0 +1,91 @@
+"""Task types and level of theory components for Q-Chem calculations"""
+
+TASK_TYPES = [
+    "single-point",
+    "geometry optimization",
+    "frequency analysis",
+    "frequency-flattening geometry optimization",
+    "transition-state geometry optimization",
+    "frequency-flattening transition-state geometry optimization",
+    "unknown"
+]
+
+FUNCTIONAL_CLASSES = {
+    "gga": [
+        "PBE",
+        # "PBE-D3(BJ)",
+        # "BLYP",
+        # "BLYP-D3(BJ)",
+        "B97-D",
+        "B97-D3",
+        # "mPW91",
+        # "mPW91-D3(BJ)",
+        # "VV10",
+        # "rVV10"
+    ],
+    "meta-gga": [
+        # "M06-L",
+        # "M06-L-D3(0)",
+        # "SCAN",
+        # "SCAN-D3(BJ)",
+        # "TPSS",
+        # "TPSS-D3(BJ)",
+        # "MN12-L",
+        # "MN12-L-D3(BJ)",
+        "B97M-rV"
+    ],
+    "hybrid-gga": [
+        # "PBE0",
+        # "PBE0-D3(BJ)",
+        "B3LYP",
+        # "B3LYP-D3(BJ)",
+        # "CAM-B3LYP",
+        # "CAM-B3LYP-D3(0)",
+        # "mPW1PW91",
+        # "mPW1PW91-D3(BJ)",
+        # "wB97X",
+        "wB97X-D",
+        "wB97X-D3",
+        "wB97X-V"
+    ],
+    "hybrid-meta-gga": [
+        # "M06-2X",
+        # "M06-2X-D3(0)",
+        # "M06-HF",
+        # "M08-SO",
+        # "M11",
+        # "MN15",
+        # "BMK",
+        # "BMK-D3(BJ)",
+        # "TPSSh",
+        # "TPSSh-D3(BJ)",
+        # "SCAN0",
+        # "mPWB1K",
+        # "mPWB1K-D3(BJ)",
+        "wB97M-V"
+    ],
+}
+
+FUNCTIONALS = [rt for functional_class in FUNCTIONAL_CLASSES for rt in FUNCTIONAL_CLASSES[functional_class]]
+
+BASIS_SETS = ["6-31g*", "def2-SVPD", "def2-TZVP", "def2-TZVPD", "def2-TZVPP", "def2-TZVPPD", "def2-QZVPPD"]
+
+SOLVENT_MODELS = ["VACUUM", "PCM", "SMD"]
```

Comment:
Happy to add ISOSVP and CMIRS, but why SM8? It's very similar to SMD (sometimes slightly better, sometimes slightly worse), but it's only parameterized (at least in Q-Chem) for a few basis sets, making it way less useful.

Reply:
I was thinking it would be valuable for completeness, if not a huge amount of extra work. I find one of the challenges of our infrastructure is that it's often built for a very specific purpose (e.g. calculation settings that the developer uses today). Then later, if someone new comes along and has a need to use different settings they have to spend significant time and effort learning and patching our code before they can use it. So when I'm writing new features I try to go the extra mile to make them as comprehensive as possible (within reasonable limits of time and energy). I figure in the long run it's much more efficient to add this support now while we have our heads in this document model than for someone to have to patch it in later.

OK, stepping off my soapbox now, in this specific case I understand the argument that SM8 is not so useful. However I don't think we can say it's obsolete either, which means someone might come along and want to use it in the future. I feel the same about solvent parameters (but appreciate that it's a lot of work to manually code them all).

I'm not going to say we _have_ to include SM8 right now if you don't feel it's a wise use of time, but the above is why I suggested it.